@copilot 

```

  Running tests on alasql@4.10.1

  Environment detected: {
    "alasqlPath": "/home/runner/_work/alasql/alasql/dist",
    "isBrowser": false,
    "isCordova": false,
    "isMeteor": false,
    "isMeteorClient": false,
    "isMeteorServer": false,
    "isNode": true,
    "isWebWorker": false
}
/home/runner/_work/alasql/alasql/test

 Exception during run: ReferenceError: n2u is not defined
    at /home/runner/_work/alasql/alasql/dist/alasql.fs.js:9490:57
    at Array.forEach (<anonymous>)
    at yy.Select.compileSelect1 (/home/runner/_work/alasql/alasql/dist/alasql.fs.js:9252:15)
    at Select.compile (/home/runner/_work/alasql/alasql/dist/alasql.fs.js:7547:27)
    at alasql.dexec (/home/runner/_work/alasql/alasql/dist/alasql.fs.js:4860:38)
    at alasql.exec (/home/runner/_work/alasql/alasql/dist/alasql.fs.js:4796:17)
    at alasql (/home/runner/_work/alasql/alasql/dist/alasql.fs.js:137:17)
    at /home/runner/_work/alasql/alasql/test/test374.js:154:21
    at Array.forEach (<anonymous>)
    at Suite.<anonymous> (/home/runner/_work/alasql/alasql/test/test374.js:140:5)
    at Object.create (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/interfaces/common.js:153:19)
    at context.describe.context.context (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/interfaces/bdd.js:46:27)
    at Object.<anonymous> (/home/runner/_work/alasql/alasql/test/test374.js:7:1)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:203:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:337:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:432:15)
    at async formattedImport (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/nodejs/esm-utils.js:10:14)
    at async Object.tryImportAndRequire [as requireOrImport] (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/nodejs/esm-utils.js:47:28)
    at async exports.loadFilesAsync (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/nodejs/esm-utils.js:152:20)
    at async singleRun (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/cli/run-helpers.js:168:3)
    at async exports.handler (/home/runner/_work/alasql/alasql/node_modules/mocha/lib/cli/run.js:379:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.

```

--- a/src/15utility.js
+++ b/src/15utility.js
@@ -1,4 +1,4 @@
-/*jshint unused:false*/
+	/*jshint unused:false*/
 /*
 	Utilities for Alasql.js
 
@@ -25,10 +25,10 @@ var utils = (alasql.utils = {});
   NaN         => undefined
 
   */
-function nanToUndefined(s) {
-	//rename for clarity.
+function nanToUndefined(s) { //rename for clarity.
 	return '(y=' + s + ',y===y?y:undefined)';
 }
+var n2u = nanToUndefined; // Alias for backward compatibility
 
 /**
   Return undefined if s undefined
@@ -42,10 +42,10 @@ function nanToUndefined(s) {
   NaN,a       => undefined
 
   */
-function undefinedOrValue(s, r) {
-	//rename for clarity
+function undefinedOrValue(s, r) { //rename for clarity
 	return '(y=' + s + ',typeof y=="undefined"?undefined:' + r + ')';
 }
+var und = undefinedOrValue; // Alias for backward compatibility
 
 /**
   Return always true. Stub for non-ecisting WHERE clause, because is faster then if(whenrfn) whenfn()
@@ -1195,54 +1195,55 @@ var domEmptyChildren = (utils.domEmptyChildren = function (container) {
 var patternCache = {};
 var like = (utils.like = function (pattern, value, escape = '') {
 	if (!patternCache[pattern]) {
-		var regexStr = '^'; // Start regex pattern to match from the beginning.
-		var i = 0; // Index for traversing the pattern string.
 
-		while (i < pattern.length) {
-			var currentChar = pattern[i];
-			var nextChar = i < pattern.length - 1 ? pattern[i + 1] : '';
+    		var regexStr = '^'; // Start regex pattern to match from the beginning.
+    		var i = 0; // Index for traversing the pattern string.
 
-			// Handle escape character.
-			if (currentChar === escape) {
-				regexStr += '\\' + nextChar;
-				i++; // Skip next character as it's escaped.
-			}
-			// Handle negation within character classes.
-			else if (currentChar === '[' && nextChar === '^') {
-				regexStr += '[^';
-				i++; // Include '^' as part of the set.
-			}
-			// Directly append square brackets.
-			else if (currentChar === '[' || currentChar === ']') {
-				regexStr += currentChar;
-			}
-			// Replace '%' with regex to match any character sequence.
-			else if (currentChar === '%') {
-				regexStr += '[\\s\\S]*';
-			}
-			// Replace '_' with regex to match any single character.
-			else if (currentChar === '_') {
-				regexStr += '.';
-			}
-			// Escape special regex characters.
-			else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {
-				regexStr += '\\' + currentChar;
-			}
-			// Append literal characters.
-			else {
-				regexStr += currentChar;
-			}
-			i++;
-		}
+   		while (i < pattern.length) {
+		      var currentChar = pattern[i];
+		      var nextChar = i < pattern.length - 1 ? pattern[i + 1] : '';
 
-		regexStr += '$'; // End regex pattern to match until the end.
-		// Compile and cache the regex pattern for future use.
-		patternCache[pattern] = RegExp(regexStr, 'i');
-	}
+   		   // Handle escape character.
+   		   if (currentChar === escape) {
+  		      regexStr += '\\' + nextChar;
+  		      i++; // Skip next character as it's escaped.
+ 		     }
+  		   // Handle negation within character classes.
+     		   else if (currentChar === '[' && nextChar === '^') {
+     		   regexStr += '[^';
+    		    i++; // Include '^' as part of the set.
+  		    }
+ 		     // Directly append square brackets.
+  		    else if (currentChar === '[' || currentChar === ']') {
+  		      regexStr += currentChar;
+ 		     }
+  		    // Replace '%' with regex to match any character sequence.
+ 		     else if (currentChar === '%') {
+  		      regexStr += '[\\s\\S]*';
+ 		     }
+		      // Replace '_' with regex to match any single character.
+		      else if (currentChar === '_') {
+		        regexStr += '.';
+		      }
+ 		     // Escape special regex characters.
+ 		     else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {
+ 		       regexStr += '\\' + currentChar;
+		      }
+ 		     // Append literal characters.
+		      else {
+		        regexStr += currentChar;
+		      }
+		      i++;
+		    }
 
-	// Convert value to string (handling null/undefined) and test against compiled pattern.
-	return ('' + (value ?? '')).search(patternCache[pattern]) > -1;
-});
+ 		   regexStr += '$'; // End regex pattern to match until the end.
+		    // Compile and cache the regex pattern for future use.
+		    patternCache[pattern] = RegExp(regexStr, 'i');
+ 		 }
+
+		  // Convert value to string (handling null/undefined) and test against compiled pattern.
+		  return ('' + (value ?? '')).search(patternCache[pattern]) > -1;
+		});
 
 /**
  * Tests if a given value matches a glob pattern.
@@ -1257,37 +1258,32 @@ utils.glob = function (value, pattern) {
 	var currentIndex = 0; // Index for traversing the pattern string.
 	var regexPattern = '^'; // Start regex pattern to match from the beginning.
 
-	while (currentIndex < pattern.length) {
+	 while (currentIndex < pattern.length) {
 		var currentChar = pattern[currentIndex];
 		var nextChar = currentIndex < pattern.length - 1 ? pattern[currentIndex + 1] : '';
-
+   		
 		// Handle character sets and negation within them.
 		if (currentChar === '[' && nextChar === '^') {
 			regexPattern += '[^';
 			currentIndex++; // Include '^' as part of the set.
-		} else if (currentChar === '[' || currentChar === ']') {
-			// Directly append square brackets.
+		} else if (currentChar === '[' || currentChar === ']') { // Directly append square brackets.
 			regexPattern += currentChar;
-		} else if (currentChar === '*') {
-			// Replace '*' with regex to match any character sequence.
+		} else if (currentChar === '*') { // Replace '*' with regex to match any character sequence.
 			regexPattern += '.*';
-		} else if (currentChar === '?') {
-			// Replace '?' with regex to match any single character.
+		} else if (currentChar === '?') { // Replace '?' with regex to match any single character.
 			regexPattern += '.';
-		} else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {
-			// Escape special regex characters.
+		}  else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {// Escape special regex characters.
 			regexPattern += '\\' + currentChar;
-		} else {
-			// Append literal characters.
+		} else { // Append literal characters.
 			regexPattern += currentChar;
 		}
 		currentIndex++;
 	}
 
-	regexPattern += '$'; // End regex pattern to match until the end.
+	 regexPattern += '$'; // End regex pattern to match until the end.
 
 	// Convert value to uppercase, compile the regex pattern in uppercase to perform a case-insensitive match.
-	return ('' + (value || '')).toUpperCase().search(RegExp(regexPattern.toUpperCase())) > -1;
+	  return ('' + (value || '')).toUpperCase().search(RegExp(regexPattern.toUpperCase())) > -1;
 };
 
 /**

--- a/src/15utility.js
+++ b/src/15utility.js
@@ -1,4 +1,4 @@
-	/*jshint unused:false*/
+/*jshint unused:false*/
 /*
 	Utilities for Alasql.js
 
@@ -25,7 +25,8 @@ var utils = (alasql.utils = {});
   NaN         => undefined
 
   */
-function nanToUndefined(s) { //rename for clarity.
+function nanToUndefined(s) {
+	//rename for clarity.
 	return '(y=' + s + ',y===y?y:undefined)';
 }
 var n2u = nanToUndefined; // Alias for backward compatibility
@@ -42,7 +43,8 @@ var n2u = nanToUndefined; // Alias for backward compatibility
   NaN,a       => undefined
 
   */
-function undefinedOrValue(s, r) { //rename for clarity
+function undefinedOrValue(s, r) {
+	//rename for clarity
 	return '(y=' + s + ',typeof y=="undefined"?undefined:' + r + ')';
 }
 var und = undefinedOrValue; // Alias for backward compatibility
@@ -1195,55 +1197,54 @@ var domEmptyChildren = (utils.domEmptyChildren = function (container) {
 var patternCache = {};
 var like = (utils.like = function (pattern, value, escape = '') {
 	if (!patternCache[pattern]) {
+		var regexStr = '^'; // Start regex pattern to match from the beginning.
+		var i = 0; // Index for traversing the pattern string.
 
-    		var regexStr = '^'; // Start regex pattern to match from the beginning.
-    		var i = 0; // Index for traversing the pattern string.
+		while (i < pattern.length) {
+			var currentChar = pattern[i];
+			var nextChar = i < pattern.length - 1 ? pattern[i + 1] : '';
 
-   		while (i < pattern.length) {
-		      var currentChar = pattern[i];
-		      var nextChar = i < pattern.length - 1 ? pattern[i + 1] : '';
+			// Handle escape character.
+			if (currentChar === escape) {
+				regexStr += '\\' + nextChar;
+				i++; // Skip next character as it's escaped.
+			}
+			// Handle negation within character classes.
+			else if (currentChar === '[' && nextChar === '^') {
+				regexStr += '[^';
+				i++; // Include '^' as part of the set.
+			}
+			// Directly append square brackets.
+			else if (currentChar === '[' || currentChar === ']') {
+				regexStr += currentChar;
+			}
+			// Replace '%' with regex to match any character sequence.
+			else if (currentChar === '%') {
+				regexStr += '[\\s\\S]*';
+			}
+			// Replace '_' with regex to match any single character.
+			else if (currentChar === '_') {
+				regexStr += '.';
+			}
+			// Escape special regex characters.
+			else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {
+				regexStr += '\\' + currentChar;
+			}
+			// Append literal characters.
+			else {
+				regexStr += currentChar;
+			}
+			i++;
+		}
 
-   		   // Handle escape character.
-   		   if (currentChar === escape) {
-  		      regexStr += '\\' + nextChar;
-  		      i++; // Skip next character as it's escaped.
- 		     }
-  		   // Handle negation within character classes.
-     		   else if (currentChar === '[' && nextChar === '^') {
-     		   regexStr += '[^';
-    		    i++; // Include '^' as part of the set.
-  		    }
- 		     // Directly append square brackets.
-  		    else if (currentChar === '[' || currentChar === ']') {
-  		      regexStr += currentChar;
- 		     }
-  		    // Replace '%' with regex to match any character sequence.
- 		     else if (currentChar === '%') {
-  		      regexStr += '[\\s\\S]*';
- 		     }
-		      // Replace '_' with regex to match any single character.
-		      else if (currentChar === '_') {
-		        regexStr += '.';
-		      }
- 		     // Escape special regex characters.
- 		     else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {
- 		       regexStr += '\\' + currentChar;
-		      }
- 		     // Append literal characters.
-		      else {
-		        regexStr += currentChar;
-		      }
-		      i++;
-		    }
+		regexStr += '$'; // End regex pattern to match until the end.
+		// Compile and cache the regex pattern for future use.
+		patternCache[pattern] = RegExp(regexStr, 'i');
+	}
 
- 		   regexStr += '$'; // End regex pattern to match until the end.
-		    // Compile and cache the regex pattern for future use.
-		    patternCache[pattern] = RegExp(regexStr, 'i');
- 		 }
-
-		  // Convert value to string (handling null/undefined) and test against compiled pattern.
-		  return ('' + (value ?? '')).search(patternCache[pattern]) > -1;
-		});
+	// Convert value to string (handling null/undefined) and test against compiled pattern.
+	return ('' + (value ?? '')).search(patternCache[pattern]) > -1;
+});
 
 /**
  * Tests if a given value matches a glob pattern.
@@ -1258,32 +1259,37 @@ utils.glob = function (value, pattern) {
 	var currentIndex = 0; // Index for traversing the pattern string.
 	var regexPattern = '^'; // Start regex pattern to match from the beginning.
 
-	 while (currentIndex < pattern.length) {
+	while (currentIndex < pattern.length) {
 		var currentChar = pattern[currentIndex];
 		var nextChar = currentIndex < pattern.length - 1 ? pattern[currentIndex + 1] : '';
-   		
+
 		// Handle character sets and negation within them.
 		if (currentChar === '[' && nextChar === '^') {
 			regexPattern += '[^';
 			currentIndex++; // Include '^' as part of the set.
-		} else if (currentChar === '[' || currentChar === ']') { // Directly append square brackets.
+		} else if (currentChar === '[' || currentChar === ']') {
+			// Directly append square brackets.
 			regexPattern += currentChar;
-		} else if (currentChar === '*') { // Replace '*' with regex to match any character sequence.
+		} else if (currentChar === '*') {
+			// Replace '*' with regex to match any character sequence.
 			regexPattern += '.*';
-		} else if (currentChar === '?') { // Replace '?' with regex to match any single character.
+		} else if (currentChar === '?') {
+			// Replace '?' with regex to match any single character.
 			regexPattern += '.';
-		}  else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {// Escape special regex characters.
+		} else if ('/.*+?|(){}'.indexOf(currentChar) > -1) {
+			// Escape special regex characters.
 			regexPattern += '\\' + currentChar;
-		} else { // Append literal characters.
+		} else {
+			// Append literal characters.
 			regexPattern += currentChar;
 		}
 		currentIndex++;
 	}
 
-	 regexPattern += '$'; // End regex pattern to match until the end.
+	regexPattern += '$'; // End regex pattern to match until the end.
 
 	// Convert value to uppercase, compile the regex pattern in uppercase to perform a case-insensitive match.
-	  return ('' + (value || '')).toUpperCase().search(RegExp(regexPattern.toUpperCase())) > -1;
+	return ('' + (value || '')).toUpperCase().search(RegExp(regexPattern.toUpperCase())) > -1;
 };
 
 /**

